chat-server/server.js

```diff
@@ -3,28 +3,92 @@ import express from "express";
 import cors from "cors";
 import path from "path";
 import { fileURLToPath } from "url";
+// Get __dirname in ES module
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
 
 const app = express();
 
 app.use(cors());
+app.use(express.urlencoded({ extended: true }));
+app.use(express.json())
+app.use(express.static(__dirname));
+
 
-// Get __dirname in ES module
-const __dirname = path.dirname(fileURLToPath(import.meta.url));
 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
 };
 
+const newMessages = [
+  {
+    id: 1,
+    from: "Peter",
+    text: "Hello, CYFers!",
+  },
+  {
+    id: 2,
+    from: "Gary",
+    text: "How are you?",
+  }
+]
+
 //This array is our "data store".
 //We will start with one message in the array.
-const messages = [welcomeMessage];
+const messages = [welcomeMessage, ...newMessages];
+
+// GET routes
 
 app.get("/", (request, response) => {
-  response.sendFile(__dirname + "/index.html");
+  response.sendFile(`${__dirname}/index.html`);
+});
+
+app.get("/messages", (request, response) => {
+  response.json(messages)
+});
+
+app.get("/messages/:messageId", (request, response) => {
+  const messageId = parseInt(request.params.messageId);
+  const message = messages.find((message) => message.id === messageId);
+  if (message) {
+    response.json(message);
+  } else {
+    response.status(404).json("Message not found");
+  }
+});
+
+// POST routes
+
+app.post("/messages", (request, response) => {
+  const newMessage = request.body;
+  const { from, text } = newMessage;
+  if (!from || !text) {
+    response.status(400).json("Message must have a Name and a Text");
+    return;
+  }
+
+  const lastId = messages[messages.length - 1].id;
+  newMessage.id = lastId + 1;
```

Comment:
It's possible this may end up re-using an ID from a previous message (e.g. if you added a message, deleted it, then added a new one). In this particular server I don't think that causes any problems, but as the application grows I can imagine it may cause problems. Can you think why it may be problematic? How could you avoid re-using IDs if you wanted to?

Reply:
Thank you.
I've made adjustments by using the reduce method to find the largest ID value and creating the new one by adding 1 to the largest.
In a real project, I think the best option would be to generate a random alphanumeric value based on the time and date of the message.